### PR TITLE
Add some coarse debug information to semantics.

### PR DESCRIPTION
**Pull request by jonmeow**
Example stack:

```
1.	node_stack_:
	0.	FunctionDefinitionStart
	1.	ReturnStatement -> node1
2.	node_block_stack_:
	0.	block0
	1.	block1
```

Example trace output:

```
*** SemanticsParseTreeHandler::Build Begin ***
Push 0: FunctionIntroducer
Push 1: DeclaredName
Push 2: ParameterListEnd
Pop 2: ParameterListEnd
Push 2: ParameterList
Pop 2: ParameterList
Pop 0: FunctionIntroducer
AddNode block0: FunctionDeclaration()
AddNode block0: BindName(ident0, node0)
AddNode block0: FunctionDefinition(node0, block1)
Push 0: FunctionDefinitionStart
Push 1: Literal -> IntegerLiteral
AddNode block1: IntegerLiteral(int0): node_xref1
Push 2: StatementEnd
Pop 2: StatementEnd
Pop 1: any (Literal) -> node0
Push 1: ReturnStatement -> ReturnExpression
AddNode block1: ReturnExpression(node0)
Pop 0: FunctionDefinitionStart
Push 0: FunctionDefinition
*** SemanticsParseTreeHandler::Build End ***
cross_reference_irs.size == 2,
cross_references = {
  node_xref0 = "xref(ir0, block0, node0)";
  node_xref1 = "xref(ir0, block0, node1)";
},
identifiers = {
  ident0 = "Foo";
},
integer_literals = {
  int0 = 0;
},
node_blocks = {
  block0 = {
    node0 = FunctionDeclaration();
    node1 = BindName(ident0, node0);
    node2 = FunctionDefinition(node0, block1);
  },
  block1 = {
    node0 = IntegerLiteral(int0): node_xref1;
    node1 = ReturnExpression(node0);
  },
}
```

### CODE REVIEW

File: toolchain/semantics/semantics_ir.cpp [hunk 1]
```diff
@@ -47,10 +47,10 @@ auto SemanticsIR::MakeBuiltinIR() -> SemanticsIR {
 auto SemanticsIR::MakeFromParseTree(const SemanticsIR& builtin_ir,
                                     const TokenizedBuffer& tokens,
                                     const ParseTree& parse_tree,
-                                    llvm::raw_ostream* verbose_stream)
+                                    llvm::raw_ostream* vlog_stream_)
```
Comment (chandlerc) [inline toolchain/semantics/semantics_ir.cpp]: Do you need to use the trailing `_` here and elsewhere in parameters? I thought the goal was to *only* support logging in class methods to start with?

Comment (jonmeow) [reply toolchain/semantics/semantics_ir.cpp]: sed mistake; fixed

File: toolchain/semantics/semantics_parse_tree_handler.h [hunk 2]
```diff
@@ -17,13 +17,22 @@ class SemanticsParseTreeHandler {
...
   explicit SemanticsParseTreeHandler(const TokenizedBuffer& tokens,
                                      const ParseTree& parse_tree,
-                                     SemanticsIR& semantics)
-      : tokens_(&tokens), parse_tree_(&parse_tree), semantics_(&semantics) {}
+                                     SemanticsIR& semantics,
+                                     llvm::raw_ostream* vlog_stream_)
+      : tokens_(&tokens),
+        parse_tree_(&parse_tree),
+        semantics_(&semantics),
+        vlog_stream_(vlog_stream_) {}
```
Comment (chandlerc) [inline toolchain/semantics/semantics_parse_tree_handler.h]: Missed `_`...
```suggestion
                                     llvm::raw_ostream* vlog_stream)
      : tokens_(&tokens),
        parse_tree_(&parse_tree),
        semantics_(&semantics),
        vlog_stream_(vlog_stream) {}
```

**Review by chandlerc — APPROVED** [review]
LG

**Pull request merged**